docs/cookbook/promotions/catalog-promotion-additional-item-filters.rst

```diff
@@ -0,0 +1,157 @@
+How to add custom filters on the item promotion actions?
+========================================================
+
+The "Item percentage discount" and "Item fixed discount" promotion actions can be used to filter which products the promotion is applied to.
+By default, you can filter by price, by taxon or by product.
+
+You can also add your own filters, for example, if you've added a brand to your products and would like to apply a discount to a particular brand.
+
+See what steps need to be taken to achieve that:
+
+Create a new item filter
+------------------------
+
+You will need a new class ``BrandFilter`` that implements the ``Sylius\Component\Core\Promotion\Filter\FilterInterface`` interface.
```

Comment:
I like the `Brand` example, but wondering if wouldn't it be better to show new custom filter on already existing functionality, for example product variant `tax category` or product inventory `tracked` property or `shipping required`.

It's just an idea for discussion (I prefer Brand example)

WDYT?

Reply:
Indeed, the product doesn't have a "brand" but I thought the example was easier to understand.

Would you like me to change the example? 
I can:
- Add the `brand` property to the Product entity
- Use the "tax category" as you suggest

Reply:
Yes, I like "brand" example as well, but code-wise it needs to be simplified, remove every line of code, which is not necessary for simple explanation of the concept and it should be good.